packages/ripple/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Variant, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+interface Options {
+  variant: Variant;
+  darkMode: boolean;
+}
+
+let globalOptions: Options = { darkMode: false, variant: Variant.Default };
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+    RIPPLE_NAMESPACE.registeredRippleElements[event.target]
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
+    return;
+  }
+
+  globalOptions = options;
+
+  // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+  RIPPLE_NAMESPACE.registeredRippleElements[node] = options;
+
+  if (!RIPPLE_NAMESPACE.setRippleListener) {
+    document.addEventListener('click', rippleEvent, { passive: true });
```

Comment:
Should we do capture mode so we get first dibs and can't be prevented?

Probably not, but wondering out loud.

Reply:
Should we support a way to remove this event listener?

Reply:
Alternatively, i can expose `removeRippleEvent` if you think that makes more sense

Reply:
Hard to predict the use case here. If we did it in the returned function we'd only be able to clean up if it was the last ripple element being removed, which takes some cost to calculate.

Maybe let's wait and see if people request it. Curious if we'll have a need for it in testing?

Reply:
SGTM